queue.go

```diff
@@ -94,11 +117,14 @@ type Queue struct {
 }
 
 // NewQueue creates and returns a new Queue structure
-func NewQueue(url string, name string, prefetchSize int, isConsumer, durable bool, jobs chan amqp.Delivery) (*Queue, error) {
+func NewQueue(name string, prefetchSize int, isConsumer, durable bool, jobs chan amqp.Delivery) (*Queue, error) {
```

Comment:
Can you refactor it to 
```golang
func NewQueue(config Config, name string, prefetchSize int, isConsumer, durable bool, jobs chan amqp.Delivery)
```
The knowledge about env variables will be still present in that module, but you will avoid the `init()` function, which is nasty https://stackoverflow.com/a/56039373/4638604

Reply:
new queue is called from the package using initially we were passing the url 
and the problem we ar fixing is to make queue package and its config  independent

Reply:
@slsyy We want to validate environment variables as early as possible, in `main()` or `init()`, because if the variable is not set the service should not start at all because of deployment and rollback. Imagine this case:
- a new ECS service task is getting deployed on AWS
- ECS triggers the healthcheck call which is `GET /ping` in our case
- if the service has no variables configured yet it should fail to start so that REST server is not running and `GET /ping` returns an error
- ECS will not start a new service then and it should trigger a deployment alarm so we know about the issue right away and can fix it

If you lazy validate the variables when they are actually needed, e.g. in the `NewQueue()`, it means that a new task (e.g. worker) will be deployed and the old one will be stopped but `NewQueue()` will fail as soon as the client calls it. However this will not trigger the deployment alarm and we will have a downtime until someone notices the issue and fix it.

This problem can be easily avoided by validating everything in `main()` but if it is a Go module the only way is to do it in `init()`, because we don't want clients to call some module-init method which would mean that they have to know something about their modules initialization stuff.

That's why I suggest to move `Unmarshal` to `init()`

Reply:
@krisowski  we are not using init 
but instead used a singleton pattern to get env variables

why we did not not use init

because as our test requires establishing actual connection to rabbitMQ , if we read env in init than it will be executed first and we will not get the chance to set env variables

singleton pattern 
the getEnv()  will be executed only once regardless of how many times it is called
also will take care of any race condition  for  **var cfg config**

Reply:
@krisowski  
why just not 
```golang
func main() {
  cfg := queue.ReadCfgFromEnv()
  q := queue.New(cfg)
}
```
It is simple and you do not have to deal with `init()` quirks. Also it is more configurable e.g. i can change some values for testing or use 2 queues in the future, if it is necessary

Reply:
after internal discussion we decided to remove that `init()` logic. Please remove that global `cfg` and refactor code to something similar like this
```golang
url, err :=  queue.ReadURLFromEnv()
// handle error or panic
q := queue.New(url, /* rest of args*/)
```